ui/breadcrumbs.md

```diff
@@ -0,0 +1,209 @@
+|`:disable_tree`|bool|`false`|explorer|Will disable all links in breadcrumbs.|
+|`:to_explorer`|string|`nil`|explorer|Link to the explorer page. Will be appended to the penultimate breadcrumb when `:disable_tree` is set.|
+|`:x_node`|string|`x_node`|explorer|Use custom node id instead of default `x_node`.|
+
+#### Breadcrumbs
+```ruby
+{
+:breadcrumbs => [
+  { :title => ... , :url => ?, :key => ? }
+  ]
+}
+```
+
+Array of hashes which sets a way in the main navigation menu:
+
+Hash is consisted of:
+
+```ruby
+{
+    :title  => _('Cloud Providers'), # Title of the breadcrumb
+    :url    => controller_url,       # URL to the controller
+    :key    => 'root',               # ID in the tree view, only explorer controllers
+}
+```
+
+### Structure
+
+#### Non-explorer controllers
+
+Non-explorer controllers should have this structure:
+
+`show_list` > `show` > `action`
+
+- On `show_list` the title is not appended, because the title from the menu path is used.
+- On `show` the `:record_info` is not use by default, because the `@title` of the page serves as a title in the last breadcrumb.
+- On `action` the `:record_info` is used to generate item breadcrumb and `@title` is used as a action breadcrumb.
+
+##### Default list
+
+`[MENU]`
+
+ex. `Compute` > `Infrastructure` > `Hosts / Nodes`
+
+When the title of the page is different from the last item in the main navigation, as the breadcrumb title the navigation title is used.
+
+##### Detail
+
+`[MENU]` > `[ITEM TITLE]`
+
+ex. `Compute` > `Infrastructure` > `Hosts / Nodes` > `17d6ec2e-0e75-45ab-95a6-642eb41855cd (Controller) (Summary)`
+
+The whole title is used (with `Summary`, `Dashboard`)
+
+##### Action
+
+`[MENU]` > `[ITEM TITLE]` > `[ACTION HEADER]`
+
+ex. `Compute` > `Infrastructure` > `Hosts / Nodes` > `17d6ec2e-0e75-45ab-95a6-642eb41855cd (Controller)` > `Info/Settings`
+
+Notice the item title is no without the appendix. That is correct.
+
+Breadcrumbs should not be deeper than this.
+
+#### Explorer controllers
+
+##### Default + detail view
+
+`[MENU]` > `[ACCORDION TITLE]` > `[CURRENT NODE TEXT]`
+
+ex. `Compute` > `Infrastructure` > `Virtual Machines` > `VMs & Templates` > `All VMs & Templates`
+
+Accordion title should lead to the tree's root.
+
+**! Breadcrumbs do not contain the whole tree path currently !**
+
+##### Action
+
+`[MENU]` > `[ACCORDION TITLE]` > `[CURRENT NODE TEXT]` > `[ACTION TITLE]`
+
+ex. `Compute` > `Infrastructure` > `Virtual Machines` > `VMs & Templates` > `3.9ocp` > `Editing Virtual Machine "3.9ocp"`
+
+As a `action title` it used either of (sorted by the highest priority):
+- `right_cell_text` as a local variable provided to breadcrumbs template (if you provide it in `replace_right_cell` method)
+- `@title_for_breadcrumbs` (non standard use)
+- `@right_cell_text` (standard use)
+- `@title` (explorer controller switched to non-explorer)
+
+Breadcrumbs should not be deeper than this.
```

Comment:
> Breadcrumbs should not be deeper than this.

Second time I see this sentense.

Reply:
Different context (non-explorers/explorers)